tests/php/core-api/wpcom-endpoints/test-class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -396,4 +409,60 @@ public function menu_item_url_data() {
 			),
 		);
 	}
+
+	/**
+	 * Tests parsing an update count.
+	 *
+	 * @param string $menu_item Menu item.
+	 * @param string $expected  Parsed menu title & count. Or not.
+	 *
+	 * @throws \ReflectionException Noop.
+	 * @dataProvider menu_item_update_data
+	 * @covers ::parse_update_count
+	 */
+	public function test_parse_update_count( $menu_item, $expected ) {
+		$class = new ReflectionClass( 'WPCOM_REST_API_V2_Endpoint_Admin_Menu' );
+
+		$prepare_menu_item_url = $class->getMethod( 'parse_update_count' );
+		$prepare_menu_item_url->setAccessible( true );
+
+		$this->assertSame(
+			$expected,
+			$prepare_menu_item_url->invokeArgs( new WPCOM_REST_API_V2_Endpoint_Admin_Menu(), array( $menu_item ) )
+		);
+	}
+
+	/**
+	 * Data provider for test_prepare_menu_item_url.
+	 *
+	 * @return \string[][]
+	 */
+	public function menu_item_update_data() {
+		return array(
+			array(
+				'No Updates here',
+				array(),
+			),
+			array(
+				'Zero updates <span class="update-plugins count-0"><span class="update-count">0</span></span>',
+				array(
+					'title' => 'Zero updates',
+				),
+			),
+			array(
+				'Finally some updates <span class="update-plugins count-5"><span class="update-count">5</span></span>',
+				array(
+					'count' => 5,
```

Comment:
What happens if we pass `count => 0` or `count => -1`...etc? Probably not a code path but best to have it covered no?

Reply:
`count => 0` is covered in the previous data set ("Zero updates"). Negative numbers will behave similarly as it only accounts for numbers greater than 0.